openedx/features/enterprise_support/middleware.py

```diff
@@ -0,0 +1,29 @@
+"""
+Middleware for the Enterprise feature.
+
+The Enterprise feature must be turned on for this middleware to have any effect.
+"""
+
+from django.core.exceptions import MiddlewareNotUsed
+
+from openedx.features.enterprise_support import api
+
+
+class EnterpriseMiddleware(object):
+    """
+    Middleware that adds Enterprise-related content to the request.
+    """
+
+    def __init__(self):
+        """
+        We don't need to use this middleware if the Enterprise feature isn't enabled.
+        """
+        if not api.enterprise_enabled():
+            raise MiddlewareNotUsed()
+
+    def process_request(self, request):
```

Comment:
Should I add a `process_response` here too, to check if the user _became_ an Enterprise learner after a request got processed, to add in the `enterprise_customer` key? Otherwise we'll need to modify the key explicitly elsewhere where they do become this. (I think the latter is better, because the former would require an extra DB call for every single response).

Reply:
@UmanShahzad I think it would make sense to delay adding the `enterprise_customer` key to the session, and only do it on demand so that we can avoid making an extra DB call by default. This is assuming that there are fewer use cases and situations where the `enterprise_customer` key is needed than use cases where it's not. It would be good to get upstream's perspective on this as well.

Reply:
Yeah the idea is to just insert it into the session whenever we do a successful `EnterpriseCustomerUser.objects.create`, which basically always takes place in edx-enterprise so the change would be needed there. That would avoid the constant DB call and limit it to only when we create a new enterprise learner. I'd love to hear @georgebabey & @douglashall 's opinions on this PR overall when they see this.

Reply:
I agree on adding it on the creation of the EnterpriseCustomerUser makes more sense versus having to create a response processor.